pcsx2/CDVD/CDVDdiscReader.cpp

```diff
@@ -270,20 +271,41 @@ static s32 DISCreadSubQ(u32 lsn, cdvdSubQ* subq)
 
 	memset(subq, 0, sizeof(cdvdSubQ));
 
-	lsn_to_msf(&subq->discM, &subq->discS, &subq->discF, lsn + 150);
+	if (src->ReadTrackSubQ(subq) != 0)
+	{
+		u8 i = strack;
+		while (i < etrack && lsn >= tracks[i + 1].start_lba)
+			++i;
+
+		lsn -= tracks[i].start_lba;
+
+		/*if (DATA(subq->ctrl))
```

Comment:
Remove code if it is not used. If you feel like it is important to a developer to know that DATA(subq->ctrl) is set for data, present it as a comment on that field.